src/core/positioning/bluetoothreceiver.cpp

```diff
@@ -163,43 +163,37 @@ void BluetoothReceiver::doConnectDevice()
 
 void BluetoothReceiver::setSocketState( const QBluetoothSocket::SocketState socketState )
 {
-  if ( mSocketState == static_cast<QAbstractSocket::SocketState>( socketState ) )
-  {
-    return;
-  }
+  emit socketStateChanged( static_cast<QAbstractSocket::SocketState>( socketState ) );
+  emit socketStateStringChanged( socketStateString() );
+}
+
+QAbstractSocket::SocketState BluetoothReceiver::socketState()
+{
+  return mSocket ? static_cast<QAbstractSocket::SocketState>( mSocket->state() ) : QAbstractSocket::UnconnectedState;
+}
 
-  switch ( socketState )
+QString BluetoothReceiver::socketStateString()
+{
+  const QAbstractSocket::SocketState currentState = socketState();
+  switch ( currentState )
   {
-    case QBluetoothSocket::SocketState::ServiceLookupState:
-    case QBluetoothSocket::SocketState::ConnectingState:
+    case QAbstractSocket::ConnectingState:
+    case QAbstractSocket::HostLookupState:
+      return tr( "Connecting…" );
+    case QAbstractSocket::ConnectedState:
+      return tr( "Successfully connected" );
+    case QAbstractSocket::UnconnectedState:
     {
-      mSocketStateString = tr( "Connecting…" );
-      break;
-    }
-    case QBluetoothSocket::SocketState::ConnectedState:
-    {
-      mSocketStateString = tr( "Successfully connected" );
-      break;
-    }
-    case QBluetoothSocket::SocketState::UnconnectedState:
-    {
-      mSocketStateString = tr( "Disconnected" );
+      QString mSocketStateString = tr( "Disconnected" );
```

Comment:
```suggestion
      QString socketStateString = tr( "Disconnected" );
```

local `variable` should not look same as member variable `mVariable`